File: backend/src/main/java/com/joengelke/shoppinglistapp/backend/model/ItemSet.java
package com.joengelke.shoppinglistapp.backend.model;

import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Getter
@Setter
@Document
public class ItemSet {

    @Id
    private String id;
    private String name;
    private List<ItemSetItem> itemList;
    private String receiptFileId = "";

    public ItemSet(String name, List<ItemSetItem> itemList, String receiptFileId) {
        this.name = name;
        this.itemList = itemList;
        this.receiptFileId = receiptFileId;
    }

    public ItemSet(){}
}

File: frontend-android/app/src/main/java/com/joengelke/shoppinglistapp/frontend/models/ItemSet.kt
package com.joengelke.shoppinglistapp.frontend.models

import kotlinx.serialization.Serializable

@Serializable
data class ItemSet (
    val id: String,
    val name: String,
    val itemList: List<ItemSetItem>,
)

@Serializable
data class ItemSetItem (
    val id: String,
    val tmpId: String, // temporary ID for frontend use only
    val name: String,
    val amount: Double,
    val unit: String
)